### PR TITLE
Trm 29986 add heartbeat async download

### PR DESCRIPTION
**Pull request by supun-ebi**

### CODE REVIEW

File: idmapping-rest/src/main/java/org/uniprot/api/idmapping/queue/AbstractIdMappingDownloadResultWriter.java [hunk 1]
```diff
@@ -122,6 +134,19 @@ public void writeResult(
         }
     }
 
+    private void updatedNoOfProcessedEntries(String jobId, int size) {
```
Comment (LeonardoGonzales) [inline idmapping-rest/src/main/java/org/uniprot/api/idmapping/queue/AbstractIdMappingDownloadResultWriter.java:137]: same comment here, using the property

File: idmapping-rest/src/test/java/org/uniprot/api/idmapping/queue/UniRefIdMappingDownloadResultWriterTest.java [hunk 2]
```diff
@@ -82,7 +103,7 @@ void getBatchStoreEntryPairIterable() {
         assertNotNull(result);
     }
 
-    @Test
+    @Test // todo test diff format
```
Comment (LeonardoGonzales) [inline idmapping-rest/src/test/java/org/uniprot/api/idmapping/queue/UniRefIdMappingDownloadResultWriterTest.java:106]: Can you please add a test for different format?

File: common-rest/src/main/java/org/uniprot/api/rest/download/AbstractDownloadResultWriter.java [hunk 3]
```diff
@@ -117,6 +126,12 @@ public void writeResult(
         }
     }
 
+    private void updatedNoOfProcessedEntries(DownloadJob downloadJob, int size) {
```
[the remark this answers is not in the document]
Comment (supun-ebi) [reply common-rest/src/main/java/org/uniprot/api/rest/download/AbstractDownloadResultWriter.java:129]: will address this on this issue https://www.ebi.ac.uk/panda/jira/browse/TRM-30403

File: common-rest/src/main/java/org/uniprot/api/common/repository/stream/store/StoreStreamer.java [hunk 4]
```diff
@@ -56,6 +55,10 @@ public Stream<String> idsStream(SolrRequest solrRequest) {
         return config.getDocumentIdStream().fetchIds(solrRequest);
     }
 
+    public long getNoOfEntries(SolrRequest solrRequest) {
```
Comment (LeonardoGonzales) [inline common-rest/src/main/java/org/uniprot/api/common/repository/stream/store/StoreStreamer.java:59]: There is a typo here, can you please change method name to getNumberOfEntries

File: common-rest/src/main/java/org/uniprot/api/rest/download/queue/AbstractMessageListener.java [hunk 5]
```diff
@@ -82,17 +81,19 @@ protected void writeResult(
...
-        log.info("Solr ids saved for job {}", jobId);
+        updateTotalEntries(downloadJob, getNoOfEntries(request));
+        log.info("Solr ids saved for job {}", downloadJob.getId());
     }
 
     protected abstract StoreRequest getStoreRequest(DownloadRequest request);
 
     protected abstract Stream<String> streamIds(DownloadRequest request);
 
+    protected abstract long getNoOfEntries(DownloadRequest downloadRequest);
```
Comment (LeonardoGonzales) [inline common-rest/src/main/java/org/uniprot/api/rest/download/queue/AbstractMessageListener.java:92]: Can you please fix the type here as well? getNumberOfEntries

File: uniprotkb-rest/src/main/java/org/uniprot/api/uniprotkb/queue/UniProtKBMessageListener.java [hunk 6]
```diff
@@ -127,6 +126,11 @@ protected Stream<String> streamIds(DownloadRequest request) {
         return service.streamIds(request);
     }
 
+    @Override
+    protected long getNoOfEntries(DownloadRequest downloadRequest) {
```
Comment (LeonardoGonzales) [inline uniprotkb-rest/src/main/java/org/uniprot/api/uniprotkb/queue/UniProtKBMessageListener.java:130]: Fix the typo here as well

File: uniref-rest/src/main/java/org/uniprot/api/uniref/queue/UniRefMessageListener.java [hunk 7]
```diff
@@ -73,6 +72,11 @@ protected Stream<String> streamIds(DownloadRequest request) {
         return service.streamIds(request);
     }
 
+    @Override
+    protected long getNoOfEntries(DownloadRequest downloadRequest) {
```
Comment (LeonardoGonzales) [inline uniref-rest/src/main/java/org/uniprot/api/uniref/queue/UniRefMessageListener.java:77]: Fix the typo here as well

**Review by LeonardoGonzales — APPROVED** [review]
Just typo to fix, Lets not merge it to main now, and wait when the config task is ready.

**Pull request closed without merging**